test/index.spec.ts

```diff
@@ -60,22 +60,44 @@ describe('resin-semver', () => {
 			expect(semver.compare('2.0.0+rev6', 'Resin OS 2.0.0+rev3')).to.equal(1);
 			expect(semver.compare('2.0.0+rev2', 'Resin OS 2.0.0+rev5')).to.equal(-1);
 			expect(semver.compare('2.0.0+rev3', '2.0.0+rev3')).to.equal(0);
+			expect(semver.compare('2.0.0+rev1', '2.0.0+rev11')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev2', '2.0.0+rev11')).to.equal(-1);
 		});
 
 		it('should correctly compare ".dev" versions', () => {
-			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(1);
+			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(-1);
 			expect(semver.compare('2.0.0', '2.0.0-dev')).to.equal(1);
-			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(-1);
+			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);// B is invalid
```

Comment:
This comment is misleading. This version is indeed technically invalid, but we do need to actively normalize this case to handle it (because this version really does exist) as if this was `2.0.0+dev`. `2.0.0` should indeed be greater, but because `.dev` is a dev version, not because `.dev` is invalid.